src/sources/splunk_hec/mod.rs

```diff
@@ -390,21 +403,24 @@ impl SplunkSource {
     }
 
     /// Authorize request
-    fn authorization(&self) -> BoxedFilter<()> {
+    fn authorization(&self) -> BoxedFilter<(Option<String>,)> {
         let valid_credentials = self.valid_credentials.clone();
         warp::header::optional("Authorization")
             .and_then(move |token: Option<String>| {
                 let valid_credentials = valid_credentials.clone();
                 async move {
                     match (token, valid_credentials.is_empty()) {
-                        (_, true) => Ok(()),
-                        (Some(token), false) if valid_credentials.contains(&token) => Ok(()),
+                        // Remove the "Splunk " prefix if present as it is not
+                        // part of the token itself
+                        (token, true) => Ok(token.map(|t| t.replacen("Splunk ", "", 1))),
+                        (Some(token), false) if valid_credentials.contains(&token) => {
+                            Ok(Some(token.replacen("Splunk ", "", 1)))
```

Comment:
This does a little more work than you want it to. `replacen` will search the entire string for the pattern instead of only searching at the start. What you want is something like:
```suggestion
                            Ok(Some(token.strip_prefix("Splunk ").map(Into::into).unwrap_or(token)))
```

Reply:
Thanks for the explanation -- added in 73168d4